komfydb/execution/aggregate.cc

```diff
@@ -0,0 +1,196 @@
+                               std::vector<Type>& groupby_types,
+                               Record* record) {
+  for (int i = 0; i < groupby_fields.size(); i++) {
+    ASSIGN_OR_RETURN(Field * field, record->GetField(groupby_fields[i]));
+    RETURN_IF_ERROR(group_id.SetField(i, field->CreateCopy()));
+  }
+  return absl::OkStatus();
+}
+
+absl::Status UpdateGroup(AggregateTuple* group,
+                         std::vector<int>& aggregate_fields,
+                         std::vector<AggregateType>& aggregate_types,
+                         Record* record) {
+  for (int i = 0; i < aggregate_fields.size(); i++) {
+    ASSIGN_OR_RETURN(Field * next_field, record->GetField(aggregate_fields[i]));
+    RETURN_IF_ERROR(group->ApplyAggregate(aggregate_types[i], i, next_field));
+  }
+  group->IncremetGroupSize();
+  return absl::OkStatus();
+}
+
+absl::Status InitializeGroup(AggregateTuple* group,
+                             std::vector<int>& aggregate_fields,
+                             std::vector<AggregateType>& aggregate_types,
+                             Record* record) {
+  for (int i = 0; i < aggregate_fields.size(); i++) {
+    ASSIGN_OR_RETURN(Field * field, record->GetField(aggregate_fields[i]));
+    switch (aggregate_types[i]) {
+      case AggregateType::COUNT: {
+        RETURN_IF_ERROR(group->SetField(i, std::make_unique<IntField>(1)));
+        break;
+      }
+      default: {
+        RETURN_IF_ERROR(group->SetField(i, field->CreateCopy()));
+        break;
+      }
+    }
+  }
+  return absl::OkStatus();
+}
+
+}  // namespace
+
+namespace komfydb::execution {
+
+Aggregate::Aggregate(std::unique_ptr<OpIterator> child,
+                     std::vector<Aggregator::AggregateType>& aggregate_types,
+                     std::vector<int>& aggregate_fields,
+                     std::vector<int>& groupby_fields, TupleDesc& tuple_desc)
+    : OpIterator(tuple_desc),
+      child(std::move(child)),
+      aggregate_types(aggregate_types),
+      aggregate_fields(aggregate_fields),
+      groupby_fields(groupby_fields) {}
+
+absl::StatusOr<std::unique_ptr<Aggregate>> Aggregate::Create(
+    std::unique_ptr<OpIterator> child,
+    std::vector<Aggregator::AggregateType>& aggregate_types,
+    std::vector<int>& aggregate_fields, std::vector<int>& groupby_fields) {
+  TupleDesc* child_tuple_desc = child->GetTupleDesc();
+  std::vector<Type> types;
+  std::vector<std::string> fields;
+  for (int i = 0; i < aggregate_types.size(); i++) {
+    AggregateType aggregate_type = aggregate_types[i];
+    Type type(Type::INT);
+    std::string name;
+    if (aggregate_type == AggregateType::NONE) {
+      ASSIGN_OR_RETURN(type,
+                       child_tuple_desc->GetFieldType(aggregate_fields[i]));
+      ASSIGN_OR_RETURN(name,
+                       child_tuple_desc->GetFieldName(aggregate_fields[i]));
+    } else {
+      name = Aggregator::AggregateTypeToString(aggregate_type);
```

Comment:
what if we have multiple aggregates of the same type? Don't we want to append column name here for example?